AgentInterfaces/MaintenanceSchedule.cs

```diff
@@ -64,11 +70,24 @@ public MaintenanceEvent(MaintenanceEvent other)
             EventId = other.EventId;
             EventType = other.EventType;
             ResourceType = other.ResourceType;
-            AffectedResources = other.AffectedResources.ToList();
+            Resources = other.Resources.ToList();
             EventStatus = other.EventStatus;
             NotBefore = other.NotBefore;
             EventSource = other.EventSource;
             DurationInSeconds = other.DurationInSeconds;
         }
+
+        public MaintenanceEvent(string eventType, string eventStatus, string eventSource)
```

Comment:
is this constructor used in testing? If so, it should not be in this class but as a factory method in the test project.